modules/blockapis/src/impl/BlockstreamApi.ts

```diff
@@ -135,6 +136,28 @@ export class BlockstreamApi implements AddressApi, UtxoApi {
     );
   }
 
+  async getTransactionIO(txid: string): Promise<TransactionIO> {
+    const tx = await this.client.get<EsploraTransaction>(`/tx/${txid}`);
+    const inputs = tx.map((body) =>
+      body.vin.map((u) => {
+        return {
+          address: u.prevout.scriptpubkey_address,
+        };
+      })
+    );
+    const outputs = tx.map((body) =>
+      body.vout.map((u) => {
+        return {
+          address: u.scriptpubkey_address,
+        };
+      })
+    );
+    return {
+      inputs,
+      outputs,
+    };
```

Comment:
TODO: verify with btc team if this is the correct data being returned

Reply:
Typically `inputs` and `outputs` would have contain `txid` and `vout`, with `address` being additional data. So for the output input section here, it'd be something like:
```return {
    txid: u.prevout.txid,
    vout: u.prevout.vout,
    address: [...],
}```
and outputs would be something like
```body.vout.map((u, vout) => {
return {
    txid,
    vout,
    address: [...],
})```

Reply:
The function here was to just fetch the IO addresses of the transaction, I didn't add additional data as it's not currently being used anywhere else (and makes this harder to mock the larger the return type is). 

I think we can add on to this interface as we find a use case for it.

Reply:
Gotcha, then yes, LGTM